### PR TITLE
improvement(model) : avoid Optional for collection types #520

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by zregvart** [comment]
@lburgazzoli codacy really has it in for you, can you rebase and add @SuppressWarnings to tests or tune the ruleset.xml so that some of those are not reported?

**Comment by jimmidyson** [comment]
@lburgazzoli Could you resolve the conflicts please?

File: controllers/src/test/java/io/syndesis/controllers/integration/online/ActivateHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import io.syndesis.model.integration.SimpleStep;
+import io.syndesis.model.integration.Step;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ActivateHandlerTest {
+
+    @Test
+    public void extractApplicationPropertiesFromIntegration() throws Exception {
```
[the remark this answers is not in the document]
Comment (lburgazzoli) [reply controllers/src/test/java/io/syndesis/controllers/integration/online/ActivateHandlerTest.java]: I tweaked this one

[the remark this answers is not in the document]
Comment (lburgazzoli) [reply controllers/src/test/java/io/syndesis/controllers/integration/online/ActivateHandlerTest.java]: This is just annoying, isn't it ? :)

**Comment by lburgazzoli** [comment]
retest this please

**Comment by lburgazzoli** [comment]
@zregvart @jimmibot @KurtStam can you please review ?

File: ruleset.xml [hunk 2]
```diff
@@ -52,7 +52,7 @@
   </rule>
   <rule ref="rulesets/java/android.xml/DoNotHardCodeSDCard">
     <priority>3</priority>
-  </rule>
+  </rule>sert
```
Comment (zregvart) [inline ruleset.xml]: I think you have a typo here

Comment (lburgazzoli) [reply ruleset.xml]: yeah definitively, going to fix it

**Comment by lburgazzoli** [comment]
@zregvart do you ming having a look again ?

**Pull request merged**